common/student/screening.ts

```diff
@@ -110,28 +116,58 @@ export async function cancelCoCReminders(student: Student) {
     await logTransaction('cocCancel', userForStudent(student), { studentId: student.id });
 }
 
-export async function updateTutorScreening(screeningId: number, data: Pick<ScreeningInput, 'comment'>, screenerId?: number) {
-    const screening = await prisma.screening.update({
-        where: {
-            id: screeningId,
-        },
-        data: {
-            comment: data.comment,
-        },
-        include: { student: true },
-    });
-    logger.info(`Screener(${screenerId}) updated tutor screening of Student(${screening.studentId})`, data);
-}
+export async function updateStudentScreening(type: StudentScreeningType, screeningId: number, data: Partial<ScreeningInput>, screenerId?: number) {
+    const screeningModel = type === 'instructor' ? prisma.instructor_screening : prisma.screening;
+    const screeningModelLabel = type === 'instructor' ? 'InstructorScreening' : 'TutorScreening';
+    const screening = await screeningModel.findFirst({ where: { id: screeningId }, include: { student: true } });
 
-export async function updateInstructorScreening(screeningId: number, data: Pick<ScreeningInput, 'comment'>, screenerId?: number) {
-    const screening = await prisma.instructor_screening.update({
-        where: {
-            id: screeningId,
-        },
+    if (screening === null) {
+        throw new Error(`${screeningModelLabel}(${screening.id}) not found`);
+    }
+
+    // We don't allow status change from success/rejection
+    const decisionTaken = screening.status === ScreeningStatus.success || screening.status === ScreeningStatus.rejection;
+    const statusChanges = data.status && screening.status !== data.status;
+    if (decisionTaken && statusChanges) {
+        throw new Error('The status of Approved/Rejected screenings cannot be changed');
```

Comment:
Could be a PrerquisiteError etc. to not treat it as an internal error but report it in the UserApp